### PR TITLE
Move deployments to a sub-directory

### PR DESCRIPTION
**Pull request by onprem**
Move all deployments related files to `deployments` sub-directory. Updated the Makefile and CI config for the new location.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+    working_directory: /go/src/github.com/observatorium/observatorium
     docker:
       - image: quay.io/coreos/jsonnet-ci
     steps:
       - checkout
       - run: make fmt
       - run: make generate-cert
-      - run: make generate tests/manifests
-      - run: git diff --exit-code -- . ':(exclude)tests/manifests/observatorium-xyz-tls-secret.yaml' ':(exclude)tests/manifests/observatorium-xyz-tls-configmap.yaml' ':(exclude)tests/manifests/observatorium-xyz-tls-dex.yaml' ':(exclude)tests/manifests/test-ca-tls.yaml'
+      - run: make generate deployments/tests/manifests
```
Comment (bwplotka) [inline .circleci/config.yml:10]: `configuration` ? (:

Comment (bwplotka) [reply .circleci/config.yml:10]: or will clash with rhobs/configuration repo?

File: Makefile [hunk 2]
```diff
@@ -8,55 +8,56 @@ CERT_DIR ?= $(TMP_DIR)/certs
 CONTROLLER_GEN ?= $(BIN_DIR)/controller-gen
 GENERATE_TLS_CERT ?= $(BIN_DIR)/generate-tls-cert
 
+DEPLOYMENTS ?= deployments
```
Comment (bwplotka) [inline Makefile:11]:
```suggestion
DEPLOYMENTS_DIR ?= deployments
```

**Review by bwplotka — APPROVED** [review]
Nice. LGTM, just some questions and suggestions. (: 

Thanks!

File: .circleci/config.yml (hunk 1 above)
Comment (onprem) [reply .circleci/config.yml:10]: I don't think a clash will happen. Also I don't have any opinions about the sub directory name as well. I will change it to `configuration` if that sounds better to you.

**Review by bwplotka — APPROVED** [review]

**Pull request merged**